### PR TITLE
Sysbox-runc changes required along the mount-hardening modifications made in sysbox-fs component

### PR DESCRIPTION
**Pull request by rodnymolina**
As part of these changes we are deferring the container registration process till the container's rootfs preparation is completed. By doing this, we are ensuring that sysbox-runc has concluded mounting all the required resources, by the time it attempts to register a container with sysbox-fs.

Signed-off-by: Rodny Molina <rmolina@nestybox.com>

### CODE REVIEW

File: libcontainer/process_linux.go [hunk 1]
```diff
@@ -547,6 +512,16 @@ func (p *initProcess) start() (retErr error) {
 			}
 			sentRun = true
 
+		case rootfsReady:
+			// Register container into sysbox-fs.
```
Comment (ctalledo) [inline libcontainer/process_linux.go:516]: "Register container with sysbox-fs"

File: libcontainer/process_linux.go [hunk 2]
```diff
@@ -547,6 +512,16 @@ func (p *initProcess) start() (retErr error) {
...
+			if err = p.register(childPid); err != nil {
```
Comment (ctalledo) [inline libcontainer/process_linux.go:517]: "Register container with sysbox-fs".

File: libcontainer/process_linux.go [hunk 3]
```diff
@@ -632,6 +607,50 @@ func (p *initProcess) start() (retErr error) {
 	return nil
 }
 
+// sysbox-runc: register the container with sysbox-fs. This must be done after
+// childPid is obtained and all container mounts are present, but before prestart
+// hooks so that sysbox-fs is ready to respond by the time the hooks run.
+func (p *initProcess) register(childPid int) error {
```
Comment (ctalledo) [inline libcontainer/process_linux.go:613]: Suggestion": "register" -> "registerWithSysboxfs"

**Review by ctalledo — APPROVED** [review]
Looks good, just a couple of minor comments.

Also, make sure you sync with the latest sysbox-runc since there are recent changes in it.

File: libcontainer/process_linux.go (hunk 2 above)
Comment (rodnymolina) [reply libcontainer/process_linux.go:517]: Done.

File: libcontainer/process_linux.go (hunk 3 above)
Comment (rodnymolina) [reply libcontainer/process_linux.go:613]: Done.

**Pull request merged**